File: log_Exp.py
def f(n): int
	print n

def p(x,n): int int
	if n == 0:
		return 1
	elif n == 1:
		return x
	else:
		r = p(x,n/2) : int
		return r*r*p(x,n%2)


b = int(raw_input("base?")) :int
e = int(raw_input("exp?")) :int
print p(b,e)

File: log_Exp.cpp
#include "supporting_libs.h"

void f(long int n)
{
    std::cout<<n;

}

long int p(long int x,long int n)
{
    if(n==0)
    {
        return 1;
    }
    else if(n==1)
    {
        return x;
    }
    else
    {
        long int r = p(x,n/2);
        return r*r*p(x,n%2);
    }

}

int main( )
{
        long int b;
        std::cout<<"base?";
        std::cin>>b;
        long int e;
        std::cout<<"exp?";
        std::cin>>e;
std::cout<<p(b,e);
 
return 0;
}

